libraries/standard/http/cbmc/include/http_cbmc_state.h

```diff
@@ -0,0 +1,31 @@
+#ifndef HTTP_CBMC_STATE_H_
+#define HTTP_CBMC_STATE_H_
+
+#include "http_client.h"
+
+/**
+ * @brief Calls malloc based on given size or returns NULL for coverage.
+ *
+ * Implementation of safe malloc which returns NULL if the requested size is 0.
+ * The behavior of malloc(0) is platform dependent.
+ * It is possible for malloc(0) to return an address without allocating memory.
+ *
+ * @brief param[in] wantedSize Requested size to malloc.
+ */
+void * safeMalloc( size_t wantedSize );
+
+/**
+ * @brief Allocate a request headers object.
+ */
+HTTPRequestHeaders_t * allocateHttpRequestHeaders();
```

Comment:
Is there any chance you will be asked to just fill in (not allocate) a header struct?  Sometimes I write these allocators to accept a pointer to a header struct, I malloc the header struct if the pointer is null.  Then regardless of whether it is a new or existing struct, I proceed to fill in values appropriately (maybe just havocing the struct).